eng/native/tryrun.cmake

```diff
@@ -47,7 +47,7 @@ elseif(EXISTS ${CROSS_ROOTFS}/boot/system/develop/headers/config/HaikuConfig.h)
   set(CLR_CMAKE_TARGET_OS haiku)
 endif()
 
-if(DARWIN)
+if(DARWIN AND NOT DEFINED ANDROID_PLATFORM)
```

Comment:
@janvorli I am not 100% sure if this is the right fix, but I had to change it to enable cross builds on MacOS i.e. to include posix semaphores and similar features.

Reply:
I think we rather need to change how we detect and set the `DARWIN` variable above so that we don't set it for cross building on macOS when the target is not macOS.
I would change the "if" above so that the check for darwin is the last one and I would add this check right before this moved check:
```cmake
elseif(EXISTS ${CROSS_ROOTFS}/etc/os-release)
  set(GLIBC_LINUX 1)
```
Instead of setting the GLIBC_LINUX, we could have just empty body of the elseif.